### PR TITLE
Admin merchants

### LINKED ISSUE #26: 24. Admin Merchants Index
24. Admin Merchants Index

As an admin,
When I visit the admin merchants index (/admin/merchants)
Then I see the name of each merchant in the system

### LINKED ISSUE #27: 25. Admin Merchant Show
25. Admin Merchant Show

As an admin,
When I click on the name of a merchant from the admin merchants index page (/admin/merchants),
Then I am taken to that merchant's admin show page (/admin/merchants/:merchant_id)
And I see the name of that merchant

### LINKED ISSUE #28: 26. Admin Merchant Update
26. Admin Merchant Update

As an admin,
When I visit a merchant's admin show page (/admin/merchants/:merchant_id)
Then I see a link to update the merchant's information.
When I click the link
Then I am taken to a page to edit this merchant
And I see a form filled in with the existing merchant attribute information
When I update the information in the form and I click ‘submit’
Then I am redirected back to the merchant's admin show page where I see the updated information
And I see a flash message stating that the information has been successfully updated.

### LINKED ISSUE #29: 27. Admin Merchant Enable/Disable
27. Admin Merchant Enable/Disable

As an admin,
When I visit the admin merchants index (/admin/merchants)
Then next to each merchant name I see a button to disable or enable that merchant.
When I click this button
Then I am redirected back to the admin merchants index
And I see that the merchant's status has changed

### LINKED ISSUE #30: 28. Admin Merchants Grouped by Status
28. Admin Merchants Grouped by Status

As an admin,
When I visit the admin merchants index (/admin/merchants)
Then I see two sections, one for "Enabled Merchants" and one for "Disabled Merchants"
And I see that each Merchant is listed in the appropriate section

### LINKED ISSUE #31: 29. Admin Merchant Create
29. Admin Merchant Create

As an admin,
When I visit the admin merchants index (/admin/merchants)
I see a link to create a new merchant.
When I click on the link,
I am taken to a form that allows me to add merchant information.
When I fill out the form I click ‘Submit’
Then I am taken back to the admin merchants index page
And I see the merchant I just created displayed
And I see my merchant was created with a default status of disabled.

### LINKED ISSUE #32: 30. Admin Merchants: Top 5 Merchants by Revenue
30. Admin Merchants: Top 5 Merchants by Revenue

As an admin,
When I visit the admin merchants index (/admin/merchants)
Then I see the names of the top 5 merchants by total revenue generated
And I see that each merchant name links to the admin merchant show page for that merchant
And I see the total revenue generated next to each merchant name

Notes on Revenue Calculation:
- Only invoices with at least one successful transaction should count towards revenue
- Revenue for an invoice should be calculated as the sum of the revenue of all invoice items
- Revenue for an invoice item should be calculated as the invoice item unit price multiplied by the quantity (do not use the item unit price)

### LINKED ISSUE #33: 31. Admin Merchants: Top Merchant's Best Day
31. Admin Merchants: Top Merchant's Best Day

As an admin,
When I visit the admin merchants index
Then next to each of the 5 merchants by revenue I see the date with the most revenue for each merchant.
And I see a label “Top selling date for <merchant name> was <date with most sales>"

Note: use the invoice date. If there are multiple days with equal number of sales, return the most recent day.

### PR DESCRIPTION
**Pull request by algibson1**
All stories under "Admin Merchants" section of user stories

Add an index page for all merchants, listing all merchants in system
Add column to merchants table, for whether merchants are enabled or disabled
Merchants are listed according to whether they are enabled or disabled
Merchants have buttons so you can either enable or disable a merchant, depending on its current status
Some CSS styling to put the merchants into tables according to their enabled or disabled status
Add edit page to update merchants
Add create page to create merchants
Create methods to calculate top merchants by revenue and their days with most sales

### CODE REVIEW

**Review by topher-nullset — APPROVED** [review]
awesome sauce!

File: app/assets/stylesheets/admin_merchants.css [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+caption.dis {
+  color: rgb(234, 10, 10);
+  font-size: 1.2rem;
+  margin-bottom: 8px
+}
+
+/* Style for the header */
+h1 {
+  text-align: center;
+}
```
Comment (pcbennett108) [inline app/assets/stylesheets/admin_merchants.css:37]: Nice table build.
Would this be reusable on Item and Invoice pages?

File: app/views/admin/merchants/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      </tr>
+    </thead>
+    <tbody>
+      <% @merchants.disabled.each do |merchant| %>
+        <tr id="dm-<%= merchant.id %>">
+          <td><%= link_to merchant.name, admin_merchant_path(merchant) %></td>
+          <td>Disabled</td>
+          <td>
+            <%= button_to "Enable Merchant", admin_merchant_path(merchant), method: :patch, params: { enabled: true } %>
+          </td>
```
Comment (pcbennett108) [inline app/views/admin/merchants/index.html.erb:55]: I had a question about where this value is stored - merchant enabled true/false.

**Review by pcbennett108 — APPROVED** [review]
Wow! Massive update.

**Pull request merged**